### DISCUSSION TITLE
print all columns setting

### DISCUSSION THREAD

**Discussion by Arengard** [ibis-project/ibis#6944] [Q&A] [answered]
i have a csv file with lots of columns, if I use following code I will get only 20 columns, how can I read all columns?

```
import ibis
from ibis.interactive import *
import numpy as np

# create a dataframe with lots of columns
df = pd.DataFrame(np.random.rand(10, 130))
df.to_csv("N.csv")
##
t = ibis.read_csv("N.csv")
t
```

**Comment by lostmygithubaccount** [comment, marked as answer]
hi @Arengard, I'm seeing all the columns after reading the CSV file. depending on how you're using Ibis (i.e. in a terminal) they may not all fit on your screen to print out; typically in VSCode or other editors you can scroll to the right to view all columns

example here showing all columns: https://gist.github.com/lostmygithubaccount/80e2e9f18adc0076b13f383c485275a8

**Reply by Arengard** [reply]
thanks so, there is some vs code setting that prevents from showing all in my case
